libs/s25main/TerrainRenderer.cpp

```diff
@@ -145,7 +144,7 @@ void TerrainRenderer::LoadTextures(const WorldDescription& desc)
                 for(unsigned i = 0; i < textures->size(); i++)
                 {
                     terrainTextures[curIdx.value].textures.push_back(
-                      libutil::dynamicUniqueCast<glArchivItem_Bitmap>(textures->release(i)).release());
+                      dynamic_cast<glArchivItem_Bitmap*>(textures->release(i).release()));
```

Comment:
do we still need to call `.release(...)`?

Reply:
isnt this a potential memory leak? if the dynamic cast returns null but we still release the pointer?

Reply:
In theory yes, however the ptr_vector would throw (or assert?) if a nullptr was added, so we assume already, that the cast is successful, which it must be.
Still found, that we can put unique_ptr directly into those vectors, so even the possible leak is now fixed.